### PR TITLE
Add support for ARM Compiler with C and CPP languages

### PR DESCRIPTION
**Pull request by fabio-porcedda**
Tested on ARM Compiler 5.05.

### CODE REVIEW

File: mesonbuild/compilers/compilers.py [hunk 1]
```diff
@@ -193,6 +200,13 @@ def is_library(fname):
                         'minsize': [],
                         }
 
+arm_buildtype_args = {'plain': [],
+                        'debug': ['-O0', '--debug'],
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__

[E127] continuation line over-indented for visual indent

File: mesonbuild/compilers/compilers.py [hunk 2]
```diff
@@ -144,6 +144,13 @@ def is_library(fname):
                               'minsize': ['/INCREMENTAL:NO', '/OPT:REF'],
                               }
 
+arm_buildtype_linker_args = {'plain': [],
+                               'debug': [],
+                               'debugoptimized': [],
+                               'release': [],
+                               'minsize': [],
+                               }
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__

[E124] closing bracket does not match visual indentation

File: mesonbuild/compilers/compilers.py [hunk 3]
```diff
@@ -193,6 +200,13 @@ def is_library(fname):
                         'minsize': [],
                         }
 
+arm_buildtype_args = {'plain': [],
+                        'debug': ['-O0', '--debug'],
+                        'debugoptimized': ['-O1', '--debug'],
+                        'release': ['-O3', '-Otime'],
+                        'minsize': ['-O3', '-Ospace'],
+                        }
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__

[E124] closing bracket does not match visual indentation

File: mesonbuild/compilers/compilers.py [hunk 4]
```diff
@@ -144,6 +144,13 @@ def is_library(fname):
                               'minsize': ['/INCREMENTAL:NO', '/OPT:REF'],
                               }
 
+arm_buildtype_linker_args = {'plain': [],
+                               'debug': [],
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/compilers.py]: __[Flake8]__

[E127] continuation line over-indented for visual indent

**Comment by jpakkane** [comment]
There is a second implementation of this coming soon. Once it is up we'll compare the approaches and decide how to proceed.

**Comment by jpakkane** [comment]
@fabio-porcedda can you test the other outstanding PR with your setup and report if it has something broken or missing for your use case?

**Comment by fabio-porcedda** [comment]
Ok.

**Comment by jpakkane** [comment]
Merged the other implementation so all new work should happen on that. Thanks for this, closing.

**Pull request closed without merging**